docs/docs/guides/build/components/state-backed-components/managing-state-in-ci-cd.md

```diff
@@ -12,145 +12,173 @@ import TabItem from '@theme/TabItem';
 
-Run this command to refresh state for all Local Filesystem components:
+## OSS deployments
 
-```bash
-dg utils refresh-defs-state --management-type LOCAL_FILESYSTEM
-```
+For OSS deployments, you'll run state refresh commands in your CI/CD pipeline before building your deployment artifacts.
 
-This command:
-1. Discovers and loads all state-backed components in your project
-2. Filters to only those using `LOCAL_FILESYSTEM` management type
-3. Fetches fresh state from each external system (Tableau, Fivetran, etc.)
-4. Writes the state to the `.local_defs_state` directory in your project
-5. Reports success or failure for each component
+### Refreshing state in CI/CD
 
-### Example: Dockerfile with state refresh
+Run state refresh commands in your CI/CD pipeline (GitHub Actions, GitLab CI, etc.) before building your deployment artifacts.
 
-Here's a Dockerfile that refreshes state during the image build process:
+#### Basic steps
 
-```dockerfile
-FROM python:3.11-slim
+1. **Install uv**: `python -m pip install uv`
+2. **Navigate to your project**: `cd path/to/your/project`
+3. **Run refresh command**: `uv run dg utils refresh-defs-state`
 
-WORKDIR /app
+#### Example: GitHub Actions workflow
 
-# Copy project files
-COPY . /app/
+```yaml
+- name: Install uv
+  run: python -m pip install uv
+  
+- name: Refresh component state
+  run: |
+    cd path/to/your/project
+    uv run dg utils refresh-defs-state
+  shell: bash
+  
+- name: Build Docker image
+  run: docker build -t my-dagster-image .
+```
 
-# Install dependencies
-RUN pip install -e .
+### Making the instance available (Versioned State Storage)
+
+If you're using Versioned State Storage, your refresh command needs access to your Dagster instance configuration.
+
+#### Requirements
+
+- Set the `DAGSTER_HOME` environment variable to point to a valid `dagster.yaml` file
+- The `dagster.yaml` must be configured with your `defs_state_storage` backend
+- Your environment must have credentials to access the storage backend (S3, GCS, etc.)
 
-# Refresh component state during build
-RUN dg utils refresh-defs-state --management-type LOCAL_FILESYSTEM
+For more information on configuring state storage, see [Configuring versioned state storage](/guides/build/components/state-backed-components/configuring-versioned-state-storage).
+
+#### Example: GitHub Actions with DAGSTER_HOME
+
+```yaml
+- name: Set up Dagster instance config
+  run: |
+    mkdir -p $HOME/dagster_home
+    echo "$DAGSTER_YAML_CONTENT" > $HOME/dagster_home/dagster.yaml
+  env:
+    DAGSTER_YAML_CONTENT: ${{ secrets.DAGSTER_YAML }}
+    
+- name: Refresh versioned state
+  run: |
+    cd path/to/your/project
+    uv run dg utils refresh-defs-state --management-type VERSIONED_STATE_STORAGE
+  env:
+    DAGSTER_HOME: $HOME/dagster_home
+    AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+    AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  shell: bash
```

Comment:
I wonder if we can use a code snippet to the actual template for ease of maintenance.